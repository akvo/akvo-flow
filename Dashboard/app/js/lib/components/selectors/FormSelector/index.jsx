import React from 'react';
import PropTypes from 'prop-types';

// eslint-disable-next-line import/no-unresolved
// import Checkbox from 'akvo-flow/components/Checkbox';

export default class FormSelector extends React.Component {
  render() {
    const { forms, onCheck } = this.props;

    return (
      <div>
        {Object.keys(forms).map(formId => (
          <div key={formId}>
            <input
              type="checkbox"
              id={formId}
              name={formId}
              defaultChecked={forms[formId].checked}
              onChange={onCheck}
              className="displayInline"
            />

            <label htmlFor={formId}>
              {forms[formId].name}
            </label>
          </div>
        ))}
      </div>
    );
  }
}

FormSelector.propTypes = {
  forms: PropTypes.any.isRequired,
  onCheck: PropTypes.func.isRequired,
};
