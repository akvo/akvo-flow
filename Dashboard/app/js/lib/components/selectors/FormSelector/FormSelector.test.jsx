import React from 'react';
import { render, cleanup } from '@testing-library/react';
import FormSelector from './index';
import '@testing-library/jest-dom/extend-expect';

// https://github.com/testing-library/react-testing-library#suppressing-unnecessary-warnings-on-react-dom-168
const originalError = console.error;
beforeAll(() => {
  console.error = (...args) => {
    if (/supports * the "act"/.test(args[0])) {
      return;
    }
    originalError.call(console, ...args);
  };
});

afterAll(() => {
  console.error = originalError;
});

// Main Tests
describe('FormSelector Tests', () => {
  afterEach(cleanup);

  it('+++ renders <snapshot> correctly', () => {
    const wrapper = render(
      <FormSelector
        forms={{ 145492013: { name: 'Handpump', checked: false } }}
        onCheck={jest.fn()}
      />
    );
    expect(wrapper.container).toMatchSnapshot();
  });
});
