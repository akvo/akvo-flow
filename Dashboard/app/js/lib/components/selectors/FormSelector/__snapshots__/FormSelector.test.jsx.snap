// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSelector Tests +++ renders <snapshot> correctly 1`] = `
<div>
  <div>
    <div>
      <input
        class="displayInline"
        id="145492013"
        name="145492013"
        type="checkbox"
      />
      <label
        for="145492013"
      >
        Handpump
      </label>
    </div>
  </div>
</div>
`;
